### ISSUE TITLE
Charts creation

### ISSUE THREAD

**Issue by gmtoglia** [HaitiData/charts-prj#9]
The first time we create a chart, the result doesn't show up properly, the second time it show up twice.

![image](https://user-images.githubusercontent.com/16922193/31193358-e14f76b4-a911-11e7-96a3-149ff86b025c.png)

**Comment by EmereArco** [comment]
Dear Gian Michele,
Thank you for your feedback. 
We are verifing the error. It seems that there problems in generating a bar chart with Chrome.
Could you please specify which browser and version are you using?
Thank you.

**Comment by gmtoglia** [comment]
Yes, my Chrome version is: Version 61.0.3163.100 (Official Build) (64-bit). In the following images you can see the behavior on Microsoft Edge and Waterfox. Thank you.

![image](https://user-images.githubusercontent.com/16922193/31237737-cef4f7e6-a9c5-11e7-95ce-2538b5dc553a.png)

**Comment by EmereArco** [comment]
Dear Gian Michele,
the issue has been solved.
Please have further test in order to check the proper functioning of the app.

**Comment by gmtoglia** [comment]
Dear Emere, the test worked very well (on microsft Edge looks different, but we can see this later). 

I found that there is a scale error when creating the image. Look what I got when I exported the above mentioned graph:

![image](https://user-images.githubusercontent.com/16922193/31289501-f21bae00-aa96-11e7-9227-3a57d1fe0f76.png)

As you can see the values are different.

**Comment by EmereArco** [comment]
Dear Gian Michele,
Have you exported the chart named "Nombre écoles pour typologie de dégât"?
I think you have choose GRADE as category, ID as quantity and then SUM as aggregation. And the export is coherent with the chart saved with the title above mentioned.
If you want to know how many records there are for each damage classes please use COUNT as aggregation type (as in the chart named "Test"). 
I hope I have been helpful.

Emere

**Comment by gmtoglia** [comment]
Yes, I did the test again and you're right, my bad sorry. Question: can we have a control on the title when saving a chart. Now I can save a chart with no title and I think this can lead to confusion. Thank you!

![image](https://user-images.githubusercontent.com/16922193/31292320-f868b38e-aaa0-11e7-835b-12eea156e856.png)

**Comment by EmereArco** [comment]
Yep!
When you choose the fields and the aggregation type a _suggested title_ is displayed below the title form.
Then if choose to **preview** the chart and you omit to fill the form, the preview will display the suggested title. 
Conversely if you fill the title form, the preview will display the title inserted in the form.
This behaviour is not implemented when you decide to **save** the chart.
If you fill the form with a title, the chart will be saved with the title you have put in, but if you omit it no title will be saved with chart. 
This behaviour can be eventually changed, saving the suggested title if no one is given. 
I leave it to you to make the decision.

I hope next training session may help in better understanding the app.

**Comment by EmereArco** [comment]
Yep!
When you choose the fields and the aggregation type a _suggested title_ is displayed below the title form.
Then if choose to **preview** the chart and you omit to fill the form, the preview will display the suggested title.
Conversely if you fill the title form, the preview will display the title inserted in the form.
This behaviour is not implemented when you decide to **save** the chart.
If you fill the form with a title, the chart will be saved with the title you have put in, but if you omit it no title will be saved with chart.
This behaviour can be eventually changed, saving the suggested title if no one is given.
I leave it to you to make the decision.

**Comment by gmtoglia** [comment]
Ok, for me we either save the suggested title, either we oblige the user to enter a title (i.e. the title will be a compulsory field -> *Title). Like a "safety" check.

**Comment by EmereArco** [comment]
I will discuss it with my colleagues next week and I'll keep you posted.

**Comment by gmtoglia** [comment]
Dear Emere, there is a bug when requesting twice the creation of a graph. In the following video you can see that:

- The user clicks on "Create graph", the lack of a "splash-screen/progress bar" makes the user close the window and try again. After the second try there are two graphs in the popup (the request was executed twice):  [video ](https://drive.google.com/file/d/0B2dBUFcFgHc8RG5DdC1ZeXZnWUE/view?usp=sharing)

**Comment by sbsimo** [comment]
Solution proposed in https://github.com/ppasq/haitidata/pull/12
Already merged and published.
Re-assigned to @gmtoglia for closing or commenting further.
